### ISSUE TITLE
bug: ion-icon not centered in ion-item

### ISSUE THREAD

**Issue by marcjulian** [ionic-team/ionic-framework#18681]
# Bug Report

**Ionic version:**

[x] **4.5.0** and **4.6.0**

**Current behavior:**

`ion-icon` is not centered in the item:
![Item-Icon-Not-Centered-V4 5 0](https://user-images.githubusercontent.com/8985933/60421397-11839780-9bea-11e9-8235-3b5cbf55fcdf.png)

**Expected behavior:**

`ion-icon` is centered in the item:
![Item-Icon-Centered-V4 4 2](https://user-images.githubusercontent.com/8985933/60421490-3972fb00-9bea-11e9-8ffd-bcbff9d786a5.png)

It still works correctly in version **4.4.2**

**Steps to reproduce:**

Create an `ion-item` with a `ion-label`, `ion-input` and `ion-icon` slotted to the **end**.

**Related code:**

```html
<ion-item>
  <ion-label position="floating">Password</ion-label>
  <ion-input type="password"></ion-input>
  <ion-icon slot="end" name="eye"></ion-icon>
</ion-item>
```

**Other information:**

**Ionic info:**

```
Ionic:

   Ionic CLI                     : 5.1.0 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.6.0
   @angular-devkit/build-angular : 0.13.9
   @angular-devkit/schematics    : 7.3.9
   @angular/cli                  : 7.3.9
   @ionic/angular-toolkit        : 1.5.1
```

**Comment by liamdebeasi** [comment]
Thanks for the issue! I am able to reproduce this issue: https://codepen.io/liamdebeasi/pen/BgrzdJ.

We will look into this!

**Comment by brandyscarney** [comment]
Thanks! This is caused by an increase in the item height due to matching the spec and the `ion-icon` having a set margin. It can be fixed by adding the following to the icon:

```
align-self: center;
```

**Comment by marcjulian** [comment]
Thanks for looking into this issue. Awesome that you already found a solution.

**Comment by chinib** [comment]
@brandyscarney align-self works. Unfortunately the detail-arrow inside an ion-item is also aligned wrongly in this use-case. 

Is there a way to add align-self: center; to the .item-detail-icon? (as it is within shadow-root)? 
Or is it possible to add align-self to the detail arrow of the ionic4 source code, or does it break something else?

**Comment by liamdebeasi** [comment]
Hi everyone,

We recently introduced a new form component architecture in Ionic 7. As part of this, we updated the internals such that this issue is resolved. We have more information available in https://ionic.io/blog/announcing-the-ionic-7-beta.

To get the new architecture you will need to use the `label` property on `ion-input` or `ion-textarea`:

```html
<ion-item>
  <ion-input label="Password" label-placement="floating" type="password"></ion-input>
  <ion-icon slot="end" name="eye"></ion-icon>
</ion-item>
```

![image](https://user-images.githubusercontent.com/2721089/217375289-f3ac8829-5732-48b9-8f06-f98316a7b1a0.png)


The work to resolve this issue has been completed, so I am going to close this. Let me know if there are any questions.
